Frameworks/HTMLStyler/HTMLStyler/Parser/Parser.swift

```diff
@@ -93,7 +93,12 @@ public struct Parser {
 
         while !scanner.isAtEnd {
             if let textString = scanner.scanUpToCharacters(from: CharacterSet(charactersIn: "<&")) {
-                resultString += textString
+                var validString: String? = nil
+                textString.utf8CString.withUnsafeBufferPointer {
+                    validString = String(validatingUTF8: $0.baseAddress!)
+                }
+
+                resultString += validString!
```

Comment:
`$0.baseAddress` is definitly pointing to a buffer since the string is assigned with a `if let` just above, so no crash to be expected there. The buffer might be empty, but it isn't an issue.
`validString` isn't a pointer anymore by the time it is forced unwrap.
The real problem here is that `String(validatingUTF8:)` doesn't do any string modification if UTF8 is invalid, if it's invalid, the init fails and return nil. The fact that it fixes our issue confuses me more than comforting me actually.